lib/reverse_string.rb

```diff
@@ -1,5 +1,16 @@
 def reverse_characters(str)
+    #create an array dividing up the string into individual characters
+    array = [str.split("")]
+    #create a holder for to-be-reversed characters
+    reverse_array = []
+    #going through each character in the array, store it in the last position of the new holder array
```

Comment:
This step–not the code but the algorithmic thinking–is right on. The problem is in the translation of this all into code.

So think back to the prompts you did before starting these exercises–specifically the one for arrays.

How do you add something to the end of an array? This should be a simple answer. Just consider an example array, like `x = [1, 2, 3]`. Now ask yourself what specific line of code you would write to push an element to the end of `x`.

Whatever your answer is, you will definitely need that functionality in your code to follow! Because your thought process here says that you'll be storing something "in the last position of the new holder array"! :dancer: 

---

As for the "going through each character in the array" idea, that is also right on. Once again, think back (or literally go back to your prompt and read your own answer) to how to do an operation to each element in an array. The code you have is not exactly like what your answer in the prompt was.

Let's start with just these two pieces. Post questions if you've got them.